tests/functionaltests/Tests/NSURLConnection.mm

```diff
@@ -87,6 +94,151 @@ - (void)connection:(NSURLConnection*)connection didFailWithError:(NSError*)e {
 
 @end
 
+// Subclass of NSURLConnectionHelper that sets its connection on a delegate queue before starting it.
+@interface NSURLConnectionTestHelper_OperationQueue : NSURLConnectionTestHelper
+@property (retain) NSOperationQueue* operationQueue;
+@end
+
+@implementation NSURLConnectionTestHelper_OperationQueue
+- (instancetype)init {
+    if (self = [super init]) {
+        self->_operationQueue = [NSOperationQueue new];
+    }
+    return self;
+}
+
+- (NSURLConnection*)createAndStartConnectionWithRequest:(NSString*)URLString {
+    NSURLConnection* connection = [self _createConnectionWithRequest:URLString];
+    [connection setDelegateQueue:_operationQueue];
+    [connection start];
+    return connection;
+}
+
+@end
+
+// Subclass of NSURLConnectionHelper that starts its connection on a separate thread's run loop.
+@interface NSURLConnectionTestHelper_RunLoop : NSURLConnectionTestHelper
+@property BOOL isStopped;
+@property (retain) NSThread* thread;
+- (void)doNothing;
+- (void)spinRunLoop;
+@end
+
+@implementation NSURLConnectionTestHelper_RunLoop
+
+- (instancetype)init {
+    if (self = [super init]) {
+        self->_thread = [[NSThread alloc] initWithTarget:self selector:@selector(spinRunLoop) object:nil];
+        [self->_thread start];
+    }
+    return self;
+}
+
+- (void)dealloc {
+    _isStopped = YES;
+}
+
+// Does nothing. Used to keep the run loop alive.
+- (void)doNothing {
+}
+
+// Keeps the run loop on the current thread spinning.
+- (void)spinRunLoop {
+    @autoreleasepool {
+        NSRunLoop* loop = [NSRunLoop currentRunLoop];
+
+        NSTimer* timer = [NSTimer timerWithTimeInterval:0.1 target:self selector:@selector(doNothing) userInfo:nil repeats:YES];
+        [loop addTimer:timer forMode:NSDefaultRunLoopMode];
+
+        while (!_isStopped) {
+            @autoreleasepool {
+                [loop runMode:NSDefaultRunLoopMode beforeDate:[NSDate distantFuture]];
+            }
+        }
+    }
+}
+
+- (NSURLConnection*)createAndStartConnectionWithRequest:(NSString*)URLString {
+    NSURLConnection* connection = [self _createConnectionWithRequest:URLString];
+    [connection performSelector:@selector(start) onThread:_thread withObject:nil waitUntilDone:NO];
+    return connection;
+}
+@end
+
+// Test to verify a request can be successfully made and a valid response and data is received.
+static void _testRequestWithURL(NSURLConnectionTestHelper* connectionTestHelper) {
```

Comment:
I considered doing parameterized tests, but since TAEF only parameterizes on Wex Strings, it felt like I was trying too hard to make it fit, so I opted for this instead.